opengrok-indexer/src/main/java/org/opengrok/indexer/history/Repository.java

```diff
@@ -579,6 +580,31 @@ protected String getRepoRelativePath(final File file)
         return filename;
     }
 
+    /**
+     * Transfers the {@code executor} output to the {@code sink}.
+     * @return the number of calls to {@code sink}
+     */
+    static int transferExecutorOutput(BufferSink sink, Executor executor)
+            throws IOException {
+        try (InputStream in = executor.getOutputStream()) {
+            byte[] buffer = new byte[8 * 1024];
+            int iterations = 0;
+            int len;
+            while ((len = in.read(buffer)) != -1) {
+                if (len > 0) {
+                    ++iterations;
+                    sink.write(buffer, 0, len);
+                }
+            }
+            return iterations;
+        }
+    }
+
+    class HistoryRevResult {
```

Comment:
does it need to be non-static? It doesn't need any state from the enclosing class

Reply:
Hmmm what does it matter for a class with no methods or initializers?

Reply:
It contains unnecessary reference to the enclosing class